File: client/src/app/_models/npc.ts
export interface NPC{
    id: number;
    name:string;
    caption: string,
    dialogue : Dialogue
}

export interface Dialogue{
id: number;
text: string;
responses: DialogueResponse[]
}

export interface DialogueResponse{
    id: number;
    text: string;
    childDialogue: Dialogue;
}

File: API/Entities/Adventure/NPC.cs
namespace API.Entities
{
    public class NPC
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Caption { get; set; }
        public  DialogueNode Dialogue { get; set; } 
    }

    public class NpcDto
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Caption { get; set; }
        public DialogueNodeDto Dialogue { get; set; }
    }
}
